src/svd/two-sided/svd.cpp

```diff
@@ -0,0 +1,132 @@
+#include "svd.hpp"
+#include <assert.h>
+#include <math.h>
+#include <algorithm>
+#include "matrix.hpp"
+#include "nsvd.hpp"
+#include "types.hpp"
+#include "util.hpp"
+
+void matrix_frobenius(matrix_t m, double* norm, double* off_norm);
```

Comment:
Please provide some explanation as to why we need this function. You can even consider adding the link to the paper in its documentation.

Reply:
Isn't it possible to combine `matrix_frobenius` and `matrix_off_frobenius` (since their internal implementation logic looks mostly the same) by just adding suitable conditions?

Reply:
I created these two functions to de-clutter the code and self document it in a way. It is just the computation of the frobenius norm of the matrix, which is essential to compute to know when to terminate the computation. I did not combine the two methods to avoid adding extra conditions or computations that are not relevant to what is needed to be computed in each case. Is there some good way to combine them and avoid unnecessary stuff?